crates/subspace-farmer/src/commitments.rs

```diff
@@ -133,6 +140,9 @@ impl Commitments {
         let piece_count = plot.piece_count();
         let mut tags_with_offset = Vec::with_capacity(TAGS_WRITE_BATCH_SIZE);
         for batch_start in (0..piece_count).step_by(PLOT_READ_BATCH_SIZE as usize) {
+            if is_stopped() {
+                return Err(CommitmentError::Stop);
+            }
```

Comment:
Sending `is_stopped` everywhere is quite invasive and doesn't fix the problem of graceful shutdown.

Think about: if recommitment happens in the background, hits this line, moves on successfully, but on the very next moment `is_stopped()` actually flips to true and application shuts down. What happens to this cycle? It will be interrupted in the middle just like if someone killed the process.

Reply:
I believe that commitments either way will be regenerated after app restart, so we just need to safely drop everything

Reply:
We are talking about graceful shutdown and I explain why it is not achieved here. Yes, we just need to safely drop everything BEFORE proceeding with whatever comes next after drop.

In addition to commitments that you noticed, plot with all supplementary databases also remain alive in this function. Any of those can be potentially corrupted by arbitrary process interruption.